src/components/tables/Store.ts

```diff
@@ -115,51 +140,83 @@ export const getInitialState = (
             );
         },
 
+        setStatsFilter: (val) => {
+            const { setStats, statsFilter } = get();
+            const filterChanged = statsFilter !== val;
+
+            if (filterChanged) {
+                set(
+                    produce((state) => {
+                        // We want to set stats back to null while loading
+                        state.stats = null;
+                        state.statsFilter = val;
+                    }),
+                    false,
+                    'Setting stats filter'
+                );
+
+                // Need to update stats if the filter changed
+                setStats();
+            }
+        },
+
         setStats: async () => {
-            const { response } = get().query;
+            const { query, statsFilter, setHydrationErrorsExist } = get();
+            const { response } = query;
             const catalogNames = flatMap(response, (data) => {
                 return data.catalog_name;
             });
 
-            if (response.length > 0) {
-                const { data, error } = await getStatsByName(catalogNames);
+            if (response && response.length > 0) {
+                const { data, error } = await getStatsByName(
+                    catalogNames,
+                    statsFilter
+                );
 
                 if (error) {
-                    const { setHydrationErrorsExist } = get();
                     setHydrationErrorsExist(true);
-                }
-
-                if (data && data.length > 0) {
-                    const statsData = {};
-                    data.forEach((datum) => {
-                        const { catalog_name } = datum;
-                        const currentStat = statsData[catalog_name];
-
-                        if (currentStat) {
-                            Object.entries(currentStat).forEach(
-                                ([key, value]) => {
-                                    if (typeof value === 'number') {
-                                        currentStat[key] =
-                                            currentStat[key] || 0;
-                                        currentStat[key] += datum[key];
+                } else if (data) {
+                    if (data.length > 0) {
+                        const statsData = {};
```

Comment:
Can we apply a `StatsSchema` type annotation here?

Reply:
Added